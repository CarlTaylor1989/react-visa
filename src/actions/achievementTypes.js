export const SET_COMPLETED_ACHIEVEMENT = 'SET_COMPLETED_ACHIEVEMENT';
export const RESTORE_ACHIEVEMENTS = 'RESTORE_ACHIEVEMENTS';

export const UPDATE_PRODUCT_ACHIEVEMENTS = 'UPDATE_PRODUCT_ACHIEVEMENTS';
export const UPDATE_PROPOSAL_ACHIEVEMENTS = 'UPDATE_PROPOSAL_ACHIEVEMENTS';
export const UPDATE_SCORED_ACHIEVEMENTS = 'UPDATE_SCORED_ACHIEVEMENTS';
export const UPDATE_SETTINGS_ACHIEVEMENTS = 'UPDATE_SETTINGS_ACHIEVEMENTS';
export const UPDATE_TIMED_ACHIEVEMENTS = 'UPDATE_TIMED_ACHIEVEMENTS';

export const RESET_CHANGED_ACHIEVEMENT = 'RESET_CHANGED_ACHIEVEMENT';
export const RESET_ACHIEVEMENTS = 'RESET_ACHIEVEMENTS';
