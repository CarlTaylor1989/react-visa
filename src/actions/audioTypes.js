export const SET_AUDIO = 'SET_AUDIO';
export const SET_PLAY_AUDIO = 'SET_PLAY_AUDIO';
export const PLAY_AUDIO = 'PLAY_AUDIO';
export const PAUSE_AUDIO = 'PAUSE_AUDIO';
export const STOP_AUDIO = 'STOP_AUDIO';
