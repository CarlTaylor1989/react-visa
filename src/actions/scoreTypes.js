
export const SET_TOTAL_SCORE = 'SET_TOTAL_SCORE';
export const SET_DISPLAYED_SCORE = 'SET_DISPLAYED_SCORE';
export const UPDATE_DISPLAYED_SCORE = 'UPDATE_DISPLAYED_SCORE';
export const SET_RUNNING_SCORE = 'SET_RUNNING_SCORE';
export const SET_RANK = 'SET_RANK';
export const SET_SCORES = 'SET_SCORES';
export const INCREASE_SCORE = 'INCREASE_SCORE';
export const DECREASE_SCORE = 'DECREASE_SCORE';

export const SET_CATEGORIES = 'SET_CATEGORIES';
export const INCREASE_CHALLENGES_COMPLETED = 'INCREASE_CHALLENGES_COMPLETED';
export const INCREASE_PERFECT_BONUS = 'INCREASE_PERFECT_BONUS';
export const INCREASE_STREAK_BONUS = 'INCREASE_STREAK_BONUS';
export const INCREASE_PRODUCTS_COMPLETED = 'INCREASE_PRODUCTS_COMPLETED';
export const INCREASE_CLIENT_NEEDS = 'INCREASE_CLIENT_NEEDS';
export const INCREASE_PROPOSALS_COMPLETED = 'INCREASE_PROPOSALS_COMPLETED';
export const INCREASE_ACHIEVEMENTS_UNLOCKED = 'INCREASE_ACHIEVEMENTS_UNLOCKED';

export const RESET_RUNNING_CATEGORIES = 'RESET_RUNNING_CATEGORIES';
export const RESET_ALL_SCORES = 'RESET_ALL_SCORES';
