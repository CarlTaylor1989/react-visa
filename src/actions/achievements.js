import * as types from './achievementTypes';

export function setCompletedAchievement(achievement) {
  return {
    type: types.SET_COMPLETED_ACHIEVEMENT,
    achievement
  };
}

export function restoreAchievements(completed, lastCompleted) {
  return {
    type: types.RESTORE_ACHIEVEMENTS,
    completed,
    lastCompleted
  };
}

export function updateProductAchievements() {
  return {
    type: types.UPDATE_PRODUCT_ACHIEVEMENTS
  };
}

export function updateProposalAchievements() {
  return {
    type: types.UPDATE_PROPOSAL_ACHIEVEMENTS
  };
}

export function updateScoredAchievements() {
  return {
    type: types.UPDATE_SCORED_ACHIEVEMENTS
  };
}

export function updateSettingsAchievements() {
  return {
    type: types.UPDATE_SETTINGS_ACHIEVEMENTS
  };
}

export function updateTimedAchievements() {
  return {
    type: types.UPDATE_TIMED_ACHIEVEMENTS
  };
}

export function resetChangedAchievement() {
  return {
    type: types.RESET_CHANGED_ACHIEVEMENT
  };
}

export function resetAchievements() {
  return {
    type: types.RESET_ACHIEVEMENTS
  };
}
