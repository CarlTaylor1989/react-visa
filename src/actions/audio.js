import * as types from './audioTypes';

export const setAudio = audioId => ({
  type: types.SET_AUDIO,
  audioId
});

export const setPlayAudio = audioId => ({
  type: types.SET_PLAY_AUDIO,
  audioId
});

export const playAudio = () => ({
  type: types.PLAY_AUDIO
});

export const pauseAudio = () => ({
  type: types.PAUSE_AUDIO
});

export const stopAudio = () => ({
  type: types.STOP_AUDIO
});
