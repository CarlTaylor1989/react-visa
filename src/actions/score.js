import * as types from './scoreTypes';

export function setTotalScore(score) {
  return {
    type: types.SET_TOTAL_SCORE,
    score
  };
}

export function setDisplayedScore() {
  return {
    type: types.SET_DISPLAYED_SCORE
  };
}

export function updateDisplayedScore() {
  return {
    type: types.UPDATE_DISPLAYED_SCORE
  };
}

export function setRunningScore(score) {
  return {
    type: types.SET_RUNNING_SCORE,
    score
  };
}

export function setRank(rank) {
  return {
    type: types.SET_RANK,
    rank
  };
}

export function setScores(running, total) {
  return {
    type: types.SET_SCORES,
    running,
    total
  };
}

export function increaseScore(score) {
  return {
    type: types.INCREASE_SCORE,
    score
  };
}

export function decreaseScore(score) {
  return {
    type: types.DECREASE_SCORE,
    score
  };
}

export function setCategories(categories) {
  return {
    type: types.SET_CATEGORIES,
    categories
  };
}

export function increaseChallengesCompleted() {
  return {
    type: types.INCREASE_CHALLENGES_COMPLETED
  };
}

export function increasePerfectBonus() {
  return {
    type: types.INCREASE_PERFECT_BONUS
  };
}

export function increaseStreakBonus() {
  return {
    type: types.INCREASE_STREAK_BONUS
  };
}

export function increaseClientNeeds() {
  return {
    type: types.INCREASE_CLIENT_NEEDS
  };
}

export function increaseProductsCompleted() {
  return {
    type: types.INCREASE_PRODUCTS_COMPLETED
  };
}

export function increaseProposalsCompleted() {
  return {
    type: types.INCREASE_PROPOSALS_COMPLETED
  };
}

export function increaseAchievementsUnlocked() {
  return {
    type: types.INCREASE_ACHIEVEMENTS_UNLOCKED
  };
}

export function resetRunningCategories() {
  return {
    type: types.RESET_RUNNING_CATEGORIES
  };
}

export function resetAllScores() {
  return {
    type: types.RESET_ALL_SCORES
  };
}
