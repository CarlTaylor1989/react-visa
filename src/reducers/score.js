import { createReducer } from 'redux-create-reducer';
import { cloneDeep } from 'lodash';
import * as types from '../actions/scoreTypes';
import { defaultRank } from '../config/ranks';
import { getRankId } from '../lib/ranks';

export const initialState = {
  total: 0,
  scoreToDisplay: 0,
  displayedScore: 0,
  running: 0,
  categories: {
    challenges: 0,
    clientNeeds: 0,
    perfectBonus: 0,
    streakBonus: 0,
    products: 0,
    proposals: 0,
    achievements: 0
  },
  rank: defaultRank,
  rankIncreaseSession: false
};

export const scoreData = createReducer(cloneDeep(initialState), {
  [types.SET_TOTAL_SCORE](state, action) {
    return {
      ...state,
      total: action.score,
      scoreToDisplay: action.score,
      displayedScore: action.score,
      rank: getRankId(action.score)
    };
  },
  [types.SET_DISPLAYED_SCORE](state) {
    return {
      ...state,
      displayedScore: state.scoreToDisplay,
      scoreToDisplay: state.total,
    };
  },
  [types.UPDATE_DISPLAYED_SCORE](state) {
    return {
      ...state,
      displayedScore: state.total
    };
  },
  [types.SET_RUNNING_SCORE](state, action) {
    return {
      ...state,
      running: action.score
    };
  },
  [types.SET_RANK](state, action) {
    return {
      ...state,
      rank: action.rank
    };
  },
  [types.SET_SCORES](state, action) {
    return {
      ...state,
      running: action.running,
      total: action.total,
      displayedScore: action.total,
      scoreToDisplay: action.total
    };
  },
  [types.INCREASE_SCORE](state, action) {
    const total = state.total + action.score;
    const newRank = getRankId(total);
    return {
      ...state,
      running: state.running + action.score,
      total,
      rank: newRank,
      rankIncreaseSession: (newRank !== state.rank) ? true : state.rankIncreaseSession
    };
  },
  [types.DECREASE_SCORE](state, action) {
    const total = state.total - action.score;
    return {
      ...state,
      running: state.running - action.score,
      total,
      rank: getRankId(total)
    };
  },
  [types.SET_CATEGORIES](state, action) {
    return {
      ...state,
      categories: {
        ...state.categories,
        ...action.categories
      }
    };
  },
  [types.INCREASE_CHALLENGES_COMPLETED](state) {
    return {
      ...state,
      categories: {
        ...state.categories,
        challenges: state.categories.challenges + 1
      }
    };
  },
  [types.INCREASE_PERFECT_BONUS](state) {
    return {
      ...state,
      categories: {
        ...state.categories,
        perfectBonus: state.categories.perfectBonus + 1
      }
    };
  },
  [types.INCREASE_STREAK_BONUS](state) {
    return {
      ...state,
      categories: {
        ...state.categories,
        streakBonus: state.categories.streakBonus + 1
      }
    };
  },
  [types.INCREASE_PRODUCTS_COMPLETED](state) {
    return {
      ...state,
      categories: {
        ...state.categories,
        products: state.categories.products + 1
      }
    };
  },
  [types.INCREASE_CLIENT_NEEDS](state) {
    return {
      ...state,
      categories: {
        ...state.categories,
        clientNeeds: state.categories.clientNeeds + 1
      }
    };
  },
  [types.INCREASE_PROPOSALS_COMPLETED](state) {
    return {
      ...state,
      categories: {
        ...state.categories,
        proposals: state.categories.proposals + 1
      }
    };
  },
  [types.INCREASE_ACHIEVEMENTS_UNLOCKED](state) {
    return {
      ...state,
      categories: {
        ...state.categories,
        achievements: state.categories.achievements + 1
      }
    };
  },
  [types.RESET_RUNNING_CATEGORIES](state) {
    return {
      ...state,
      running: initialState.running,
      categories: {
        ...initialState.categories
      }
    };
  },
  [types.RESET_ALL_SCORES]() {
    return {
      total: initialState.total,
      displayedScore: initialState.displayedScore,
      scoreToDisplay: initialState.scoreToDisplay,
      running: initialState.running,
      rank: initialState.rank,
      categories: {
        ...initialState.categories
      },
      rankIncreaseSession: initialState.rankIncreaseSession
    };
  }
});

export default { scoreData };
