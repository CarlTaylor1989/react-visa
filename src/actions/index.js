import { push } from 'connected-react-router';
import * as AchievementActions from './achievements';
import * as AudioActions from './audio';
import * as ClientActions from './client';
import * as DiagnosticActions from './diagnostic';
import * as GenericActions from './generic';
import * as HintsActions from './hints';
import * as MapActions from './map';
import * as PromptActions from './prompts';
import * as ScoreActions from './score';
import * as ScormActions from './scorm';
import * as SettingsActions from './settings';
import * as StreakActions from './streak';
import * as TutorialActions from './tutorial';

const ActionCreators = Object.assign(
  {},
  AchievementActions,
  AudioActions,
  ClientActions,
  DiagnosticActions,
  GenericActions,
  HintsActions,
  MapActions,
  PromptActions,
  ScoreActions,
  ScormActions,
  SettingsActions,
  StreakActions,
  TutorialActions,
  { push }
);

export default ActionCreators;
