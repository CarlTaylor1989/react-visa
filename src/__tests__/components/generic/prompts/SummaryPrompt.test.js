/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import {
  SummaryPrompt,
  mapStateToProps,
  mapDispatchToProps
} from '../../../../components/generic/prompts/SummaryPrompt';
import * as actions from '../../../../actions/score';
import * as types from '../../../../actions/scoreTypes';

jest.useFakeTimers();

let mockProps;
beforeEach(() => {
  mockProps = {
    categories: {
      achievements: 0,
      products: 1
    },
    closeSummary: jest.fn(),
    resetRunningCategories: jest.fn(),
    resetLastProductCompleted: jest.fn(),
    resetChangedAchievement: jest.fn(),
    running: 10000,
    setDisplayedScore: jest.fn(),
    setPlayAudio: jest.fn()
  };
});

test('renders the SummaryPrompt component', () => {
  const wrapper = shallow(<SummaryPrompt {...mockProps} />, {
    disableLifecycleMethods: true,
  });
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('method componentDidMount()', () => {
  // setup
  const wrapper = shallow(<SummaryPrompt {...mockProps} />, {
    disableLifecycleMethods: true,
  });
  wrapper.instance().closeSummary = jest.fn();

  // function under test
  wrapper.instance().componentDidMount();
  jest.runOnlyPendingTimers();

  // expectations
  expect(wrapper.instance().closeSummary).toHaveBeenCalled();
  expect(mockProps.setDisplayedScore).toHaveBeenCalled();
});

test('method componentWillUnmount()', () => {
  // setup
  const wrapper = shallow(<SummaryPrompt {...mockProps} />, {
    disableLifecycleMethods: true,
  });

  // function under test
  wrapper.instance().componentWillUnmount();

  // expectations
  expect(mockProps.resetRunningCategories).toHaveBeenCalled();
  expect(mockProps.resetLastProductCompleted).toHaveBeenCalled();
  expect(mockProps.resetChangedAchievement).toHaveBeenCalled();
});

test('method closeSummary()', () => {
  const wrapper = shallow(<SummaryPrompt {...mockProps} />, {
    disableLifecycleMethods: true,
  });
  wrapper.instance().closeSummary();
  expect(mockProps.closeSummary).toHaveBeenCalled();
});

test('should mapStateToProps', () => {
  // setup
  const state = {
    scoreData: {
      categories: {
        foo: 0,
        bar: 1
      },
      running: 15000
    }
  };
  const expected = {
    ...state.scoreData
  };

  // function under test
  const props = mapStateToProps(state);

  // expectation
  expect(props).toEqual(expected);
});

test('should mapDispatchToProps', () => {
  const dispatch = jest.fn();
  jest.spyOn(actions, 'resetRunningCategories').mockReturnValue('resetRunningCategories()');
  const props = mapDispatchToProps(dispatch);
  props.resetRunningCategories();
  expect(dispatch).toHaveBeenCalledWith({
    type: types.RESET_RUNNING_CATEGORIES
  });
});
