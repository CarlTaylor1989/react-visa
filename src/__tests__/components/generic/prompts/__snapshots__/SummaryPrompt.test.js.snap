// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SummaryPrompt component 1`] = `
<div
  className="summaryPrompt"
>
  <div
    className="runningPoints"
  >
    <Translate
      tag="span"
      value="generic.prompts.pointsPrefix"
    />
    <CountUp$$1
      decimal="."
      decimals={0}
      delay={0}
      duration={4}
      easingFn={null}
      end={0}
      formattingFn={null}
      onEnd={[Function]}
      onPauseResume={[Function]}
      onReset={[Function]}
      onStart={[Function]}
      onUpdate={[Function]}
      prefix=""
      preserveValue={false}
      redraw={false}
      separator=","
      start={0}
      suffix=""
      useEasing={true}
    />
  </div>
  <ul
    className="categories"
  >
    <SummaryPromptCategory
      category="products"
      counter={1}
      key="products1"
    />
  </ul>
</div>
`;
